src/psyclone/psyir/nodes/call.py

```diff
@@ -593,3 +610,188 @@ def _location_txt(node):
             f"Failed to find a Routine named '{rsym.name}' in "
             f"{_location_txt(root_node)}. This is normally because the routine"
             f" is within a CodeBlock.")
+
+    def _check_argument_type_matches(
+                self,
+                call_arg: DataSymbol,
+                routine_arg: DataSymbol,
+            ) -> bool:
```

Comment:
I don't think this routine needs a return value. (Also please update the docstring accordingly.)

Reply:
I changed it to "-> None".
The :rtype: should be covered by the "-> None".
Do we still need to write a Docstring for return types?

Reply:
Not if it hasn't got one, no.

Reply:
(You can remove the `return` statement altogether too)

Reply:
OK